src/kdalgorithms.h

```diff
@@ -238,6 +238,28 @@ std::optional<Item> max_value_less_than(const Container &container, Item &&needl
     return *it;
 }
 
+template <typename Container, typename Item, typename Compare = std::less<ValueType<Container>>>
+#if __cplusplus >= 202002L
+    requires BinaryPredicateOnContainerValues<Compare, Container>
+    && ContainerOfType<Container, Item>
+#endif
+    std::optional<Item> max_value_less_than_unordered(const Container &container, Item &&needle,
```

Comment:
I'd make `needle` a `const&` - we don't want to allow the uses to do strange things by changing the needle in their comparison functions.